src/PerimeterxFirstPartyClient.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Perimeterx;
+
+final class PerimeterxFirstPartyClient {
+    const CLIENT_DOMAIN = 'client.perimeterx.net';
+    const CAPTCHA_DOMAIN = 'captcha.px-cdn.net';
+    const XHR_DOMAIN = 'collector.perimeterx.net';
+    const FIRST_PARTY_HEADER_NAME = 'X-PX-First-Party';
+    const FIRST_PARTY_HEADER_VALUE = '1';
+
+    const EMPTY_GIF_B64 = 'R0lGODlhAQABAPAAAAAAAAAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==';
+
+    private $pxConfig;
+    private $logger;
+    private $httpClient;
+
+    /**
+     * @param array $pxConfig
+     * @param Perimeterx\Utils\IHttpClient $httpClient
+     */
+
+    public function __construct($pxConfig, $httpClient) {
+        $this->pxConfig = $pxConfig;
+        $this->logger = $pxConfig['logger'];
+        $this->httpClient = $httpClient;
+    }
+
+    public function handleFirstParty() {
+        $appIdSubstr = substr($this->pxConfig['app_id'], 2);
+        $initJsPath = "/$appIdSubstr/init.js";
+        $xhrPath = "/$appIdSubstr/xhr";
+        $captchaJsPath = "/$appIdSubstr/captcha";
+
+        $uri = strtok($_SERVER['REQUEST_URI'], "?");
+        $method = $_SERVER['REQUEST_METHOD'];
+
+        $retval = null;
+
+        if ($uri === $initJsPath) {
```

Comment:
is this (=== operator) a valid comparison of 2 strings? isn't this strcmp/strpos?
https://www.tutorialrepublic.com/faq/how-to-compare-two-strings-in-php.php#:~:text=You%20can%20use%20the%20PHP,0%20if%20they%20are%20equal.

Reply:
It's possible to use the === operator as well: https://stackoverflow.com/questions/80646/how-do-the-php-equality-double-equals-and-identity-triple-equals-comp

Reply:
Okie Dokie!